### PR TITLE
Support docker-py > 1.10.6

### PR DESCRIPTION
**Pull request by anish**
This has been renamed to docker in pypi, and provides a backwards compatible API

### CODE REVIEW

File: master/buildbot/worker/docker.py [hunk 1]
```diff
@@ -201,7 +202,10 @@ def _image_exists(self, client, name):
         return False
 
     def _thd_start_instance(self, image, dockerfile, volumes):
-        docker_client = client.Client(**self.client_args)
+        if docker.version[0] == '1':
```
Comment (tardyp) [inline master/buildbot/worker/docker.py]: You should factorize this in getDockerClient:

File: master/buildbot/test/fake/docker.py [hunk 2]
```diff
@@ -82,3 +84,7 @@ def create_container(self, image, *args, **kwargs):
 
     def remove_container(self, id, **kwargs):
         del self._containers[id]
+
+
+class APIClient(Client):
```
Comment (tardyp) [inline master/buildbot/test/fake/docker.py]: I dont think it is really needed

File: master/buildbot/worker/docker.py (hunk 1 above)
Comment (anish) [inline master/buildbot/worker/docker.py]: sounds good, will do

File: master/buildbot/test/fake/docker.py (hunk 2 above)
Comment (anish) [inline master/buildbot/test/fake/docker.py]: mostly added as a test requirement, I will double check it if it's still needed after refactorization

**Comment by anish** [comment]
changes made, github doesn't pick up force push

File: master/buildbot/test/fake/docker.py (hunk 2 above)
Comment (anish) [inline master/buildbot/test/fake/docker.py]: yep, still needed for tests

**Pull request merged**